task-sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -272,7 +273,12 @@ class ConnectionAccessor:
     """Wrapper to access Connection entries in template."""
 
     def __getattr__(self, conn_id: str) -> Any:
-        return _get_connection(conn_id)
+        try:
+            return _get_connection(conn_id)
+        except AirflowRuntimeError as e:
+            if e.error.error == ErrorType.CONNECTION_NOT_FOUND:
+                raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
```

Comment:
```suggestion
                raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined") from None
```
As I don't think we want the original exec to be recorded as the cause here

Reply:
Makes sense, let me make this change everywhere.

Reply:
Yeah its much better now, thanks

<img width="1726" alt="Image" src="https://github.com/user-attachments/assets/d6a1a768-0d00-4cd8-ba49-cc76175a2cd7" />

Reply:
Pushed in 01b794136c